src/psyclone/psyir/nodes/structure_reference.py

```diff
@@ -241,8 +264,9 @@ def get_signature_and_indices(self):
     @property
     def datatype(self):
         '''
-        Walks down the list of members making up this reference to determine
-        the type that it refers to.
+        If no data type was enforced for this reference, walk down the list
```

Comment:
To make it less visible maybe leave this as it was and add the addendum after the first sentence instead.

"If a overwrite datatype was given to this reference, this will return it instead of determining the type."

Reply:
Done.